### PR TITLE
Issue #3132494 by chmez: Add ability to disable posts on streams with permissions

### PR DESCRIPTION
**Pull request by ghost**
## Problem
Remove ability to post from stream on profile

## Solution
- Create permissions for creating each post type
- Give access to creating posts of specific types
- Integrate new permissions in the access handler
- Add checking new permission in the post block for groups

## Issue tracker
- https://www.drupal.org/project/social/issues/3132494
- https://getopensocial.atlassian.net/browse/YANG-2737

## Release notes
Add permissions for managing who can create each type of posts

### CODE REVIEW

**Review by ribel — APPROVED** [review]
@OH17L Code changes looks good to me. 
Also, I tested and all works as expected.
But please update Solution section here and in an Issue to reflect actual code changes (you may copy it from your commit messages)

**Comment by OH17L** [comment]
@ribel, the **Solution** pull request section has been updated.

**Review by ribel — APPROVED** [review]

File: modules/social_features/social_post/src/Plugin/Block/PostGroupBlock.php [hunk 1]
```diff
@@ -33,14 +53,20 @@ protected function blockAccess(AccountInterface $account) {
     $group = _social_group_get_current_group();
 
     if (is_object($group)) {
-      if ($group->hasPermission('add post entities in group', $account) && $account->hasPermission("add post entities")) {
+      if (
+        $group->hasPermission('add post entities in group', $account) &&
+        (
+          $account->hasPermission('add post entities') ||
+          $account->hasPermission("add $this->bundle post entities")
```
Comment (ronaldtebrake) [inline modules/social_features/social_post/src/Plugin/Block/PostGroupBlock.php:60]: will this work?

`account->hasPermission("add $this->bundle post entities")`

Comment (OH17L) [reply modules/social_features/social_post/src/Plugin/Block/PostGroupBlock.php]: Done

**Comment by ribel** [comment]
🍒cherry-picked to 8.x-8.x

**Pull request merged**